tendrl/node_agent/node_sync/check_all_managed_nodes_status.py

```diff
@@ -1,11 +1,18 @@
-def run():
-    nodes = NS.tendrl.objects.Node().load_all()
-    if nodes is None:
-        return
+import etcd
+
+from tendrl.commons.utils import etcd_utils
 
-    for node in nodes:
-        _node_context = NS.tendrl.objects.NodeContext(
-            node_id=node.node_id
-        ).load()
-        _node_context.watch_attrs()
```

Comment:
When node_agent starts it does not have fqdn in node_context, after SDS detection and only it will have fqdn, the problem with this approach is it creates a watchable thread with node_context object which does not all the details. Whenever it calls on change function it will override the node_context detail with none values. That is why sometimes we saw fqdn None problem in gluster-integration sync.